clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -584,6 +584,9 @@ public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception
         Map<TopicPartition, OffsetCommitRequest.PartitionData> offsetData = new HashMap<>(offsets.size());
         for (Map.Entry<TopicPartition, OffsetAndMetadata> entry : offsets.entrySet()) {
             OffsetAndMetadata offsetAndMetadata = entry.getValue();
+            if (offsetAndMetadata.offset() < 0) {
+                return RequestFuture.failure(new IllegalArgumentException("Invalid offset: " + offsetAndMetadata.offset()));
```

Comment:
Do you think it makes sense to try and catch this earlier (like in `KafkaConsumer`), or did you want to ensure that the autocommit path is covered as well?